sbi/samplers/mcmc/init_strategy.py

```diff
@@ -44,8 +46,41 @@ def sir(
 
     See Rubin 1988, "Using the sir algorithm to simulate posterior distributions."
 
-    This function can also do `SIR` on the conditional posterior
-    $p(\theta_i|\theta_j, x)$ when a `condition` and `dims_to_sample` are passed.
+    Args:
+        proposal: Proposal distribution, candidate samples are drawn from it.
+        potential_fn: Potential function that the candidate samples are weighted with.
+            Note that the function needs to return log probabilities.
+        sir_num_batches: Number of candidate batches drawn.
+        sir_batch_size: Batch size used for evaluating candidates.
+
+    Returns:
+        A single sample.
+    """
+    sample = sampling_importance_resampling(
+        potential_fn=potential_fn,
+        proposal=proposal,
+        num_samples=1,
+        oversampling_factor=sir_num_batches * sir_batch_size,
+        max_sampling_batch_size=sir_batch_size,
+    )
+    return transform(sample)  # type: ignore
+
+
+def resample_given_potential_fn(
+    proposal: Any,
+    potential_fn: Callable,
+    transform: torch_tf.Transform,
+    sir_num_batches: int = 10,
+    sir_batch_size: int = 1000,
+    **kwargs: Any,
+) -> Tensor:
+    r"""Return a sample via resampling proposal samples with `potential_fn` weights.
+
+    The difference to actually performing SIR is that the weights are given only
```

Comment:
I guess we initially called this SIR as well because we assumed `proposal=prior`. But good catch!

Reply:
Sure, but even with `proposal=prior` one still has to correct. E.g. if the `prior` is gaussian, the weights should be `q(theta | x) / p(theta)` instead of just `q(theta | x)`

Reply:
I agree, and then if you have `q=potential_fn` then it cancels out, no?